zephyr/include/sof/lib/dma.h

```diff
@@ -204,8 +204,20 @@ struct dma {
 	void *priv_data;
 };
 
+/*
+ * Note: typedef use like this is generally avoided in SOF,
+ *       but this is an exceptional case to handle SOF platforms
+ *       that can be built with both Zephyr and XTOS drivers (like imx8).
+ *       Once no users left, this can be removed.
+ */
+#if CONFIG_ZEPHYR_NATIVE_DRIVERS
+typedef struct sof_dma sof_dma;
+#else
+typedef struct dma sof_dma;
+#endif
+
```

Comment:
OK I admit this is starting to get (too?) ugly. This is really only needed for the few cases where SOF Zephyr is built with using XTOS drivers. One alternative is to convert the naming also in all XTOS code, but it is quite massive amount of change as this affects most drivers and most platform code.

Reply:
Its going to be short term, so will be removed at some point.